corehq/form_processor/management/commands/archive_all_forms_for_user_in_domain.py

```diff
@@ -66,15 +68,39 @@ def handle(self, user_id, domain, **options):
                         case_ids_log.write("%s\n" % ledger_case_id)
                     LedgerAccessorSQL.delete_ledger_transactions_for_form(ledger_case_ids, xform.form_id)
 
-        # rebuild cases
-        print("Starting with cases rebuild")
-        reason = "User %s forms archived for domain %s by system" % (user.raw_username, domain)
-        form_processor_interface = FormProcessorInterface(domain)
+    def _rebuild_cases(self):
+        user = CouchUser.get_by_user_id(self.user_id)
+        reason = "User %s forms archived for domain %s by system" % (user.raw_username, self.domain)
+        form_processor_interface = FormProcessorInterface(self.domain)
         with open("cases_rebuilt.txt", "w+b") as case_log:
-            for case_id in with_progress_bar(case_ids_to_rebuild):
+            for case_id in with_progress_bar(self.case_ids_to_rebuild):
                 case_log.write("%s\n" % case_id)
-                rebuild_case_from_forms(domain, case_id, RebuildWithReason(reason=reason))
+                rebuild_case_from_forms(self.domain, case_id, RebuildWithReason(reason=reason))
                 ledgers = form_processor_interface.ledger_db.get_ledgers_for_case(case_id)
                 for ledger in ledgers:
                     form_processor_interface.ledger_processor.rebuild_ledger_state(
                         case_id, ledger.section_id, ledger.entry_id)
+
+    def handle(self, user_id, domain, **options):
+        self.user_id = user_id
+        self.domain = domain
+
+        self.forms = self._get_forms_to_archive()
+        print("Found %s normal forms for user" % len(self.forms))
+
+        self.case_ids_to_rebuild = self._fetch_case_ids_to_rebuild()
+        print("Found %s cases that would need to be rebuilt" % len(self.case_ids_to_rebuild))
+
+        # archive forms
+        print("Starting with form archival")
+        self._archive_forms()
+
+        # removing ledger transactions
+        print("Starting with removing ledger transactions")
+        self._remove_ledger_transactions()
+
+        # rebuild cases
+        print("Starting with cases rebuild")
+        self._rebuild_cases()
+
+        print("Completed!")
```

Comment:
Nice, this reads much more easily!  You could drop the comments too, now that they're redundant with the method names.